dashboard/app/controllers/application_controller.rb

```diff
@@ -179,22 +179,6 @@ def milestone_response(options)
       end
     end
 
-    unless options[:solved?]
-      # Call method to generate hint and related attributes, copying results into response.
-      hint_details = ExperimentActivity.determine_hint({
-                                                         level_source: options[:level_source],
-                                                         current_user: current_user,
-                                                         enable_external_hints: Rails.env.production?,
-                                                         ip: request.remote_ip,
-                                                         uri: request.referer,
-                                                         activity: options[:activity]})
-      response[:hint] = hint_details[:hint] if hint_details[:hint]
-      response[:hint_request_placement] = hint_details[:hint_request_placement] if
-        hint_details[:hint_request_placement]
-      response[:hint_requested_url] =
-        activity_hint_path(hint_details[:activity_hint]) if hint_details[:activity_hint]
-    end
```

Comment:
Though I have done a best effort confirming that these are never referenced, this should be confirmed rather carefully.